drivers/sms_ser.c

```diff
@@ -0,0 +1,701 @@
+        upsdebugx(3, "command test.battery.start.quick OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    if (!strcasecmp(cmdname, "test.battery.stop")) {
+        length = sms_prepare_cancel_test(&bufOut[0]);
+
+        if (ser_send_buf(upsfd, bufOut, length) == 0) {
+            upsdebugx(3, "failed to send test.battery.stop");
+            return STAT_INSTCMD_FAILED;
+        }
+        upsdebugx(3, "command test.battery.stop OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    if (!strcasecmp(cmdname, "beeper.toggle")) {
+        length = sms_prepare_set_beep(&bufOut[0]);
+
+        if (ser_send_buf(upsfd, bufOut, length) == 0) {
+            upsdebugx(3, "failed to send beeper.toggle");
+            return STAT_INSTCMD_FAILED;
+        }
+        upsdebugx(3, "command beeper.toggle OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    if (!strcasecmp(cmdname, "shutdown.return")) {
+        length = sms_prepare_shutdown_restore(&bufOut[0]);
+
+        if (ser_send_buf(upsfd, bufOut, length) == 0) {
+            upsdebugx(3, "failed to send shutdown.return");
+            return STAT_INSTCMD_FAILED;
+        }
+        upsdebugx(3, "command shutdown.return OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    if (!strcasecmp(cmdname, "shutdown.reboot")) {
+        uint16_t delay = bootdelay;
+        if (extra) {
+            long ldelay = strtol(extra, NULL, bootdelay);
+            if (ldelay >= 0 && (intmax_t)ldelay < (intmax_t)UINT16_MAX) {
+                delay = (uint16_t)ldelay;
+            } else {
+                upsdebugx(3, "tried to set up extra shutdown.reboot delay ut it was out of range, keeping default");
+            }
+        }
+        length = sms_prepare_shutdown_nsec(&bufOut[0], delay);
+
+        if (ser_send_buf(upsfd, bufOut, length) == 0) {
+            upsdebugx(3, "failed to send shutdown.reboot");
+            return STAT_INSTCMD_FAILED;
+        }
+        upsdebugx(3, "command shutdown.reboot OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    if (!strcasecmp(cmdname, "shutdown.stop")) {
+        length = sms_prepare_cancel_shutdown(&bufOut[0]);
+
+        if (ser_send_buf(upsfd, bufOut, length) == 0) {
+            upsdebugx(3, "failed to send shutdown.stop");
+            return STAT_INSTCMD_FAILED;
+        }
+        upsdebugx(3, "command shutdown.stop OK!");
+
+        return STAT_INSTCMD_HANDLED;
+    }
+
+    upslogx(LOG_NOTICE, "sms_instcmd: unknown command [%s]", cmdname);
+    return STAT_INSTCMD_UNKNOWN;
+}
+
+static int sms_setvar(const char *varname, const char *val) {
+    if (!strcasecmp(varname, "ups.delay.reboot")) {
+        int ipv = atoi(val);
+        if (ipv >= 0)
+            bootdelay = (unsigned int)ipv;
+        dstate_setinfo("ups.delay.reboot", "%u", bootdelay);
+        return STAT_SET_HANDLED;
+    }
+    return STAT_SET_UNKNOWN;
+}
+
+void upsdrv_initinfo(void) {
+    char *battery_status;
+
+	upsdebugx(LOG_DEBUG, "upsdrv_initinfo");
```

Comment:
Here and elsewhere, `upsdebugx()` first argument is a number - relatively arbitrarily assigned by driver authors, larger numbers are less visible (need higher verbosity level). Method entry is usually 2-3 or higher, and traces inside methods may be same or +1 usually...

Reply:
The LOG_DEBUG is not a const to a number ?  like:
LOG_DEBUG = 3 ?

Reply:
![Captura de tela de 2023-10-08 11-56-59](https://github.com/networkupstools/nut/assets/2652837/d664e6fd-7fd6-4ea8-97f3-81ae2c4ef8bb)

Reply:
Well, it is, but in a different "dimension" or unit of measurement. These `LOG*` definitions originate from syslog, I think (e.g. `/usr/include/x86_64-linux-gnu/sys/syslog.h` or our shims for WIN32 builds), not from NUT verbosity... traditions.

I agree it would have been useful to have symbolic names for those levels in NUT, but so far they fared well enough with the inconsistent rogue use like what we have, for a couple of decades and counting.